src/gui/ExternalToolInterface.cc

```diff
@@ -86,18 +90,66 @@ bool OctoPrintService::process(const std::string& displayName, std::function<boo
   return true;
 }
 
+QDir LocalProgramService::getTempDir() const
+{
+  const auto& tempDirConfig = Settings::Settings::localAppTempDir.value();
+  if (tempDirConfig.empty()) {
+    return QDir::temp();
+  }
+  const auto tempDir = QDir{QString::fromStdString(tempDirConfig)};
+  if (!tempDir.exists()) {
+    LOG(message_group::Warning, "Configured temporary directory does not exist: '%1$s'", tempDirConfig);
+    return QDir::temp();
+  }
+  return tempDir;
+}
+
 bool LocalProgramService::process(const std::string& displayName, std::function<bool (double)> progress_cb) 
 {
   QProcess process;
   process.setProcessChannelMode(QProcess::MergedChannels);
 
-  const QString slicer = QString::fromStdString(Settings::Settings::localSlicerExecutable.value());
+  const QString application = QString::fromStdString(Settings::Settings::localAppExecutable.value());
+
+  if (application.trimmed().isEmpty()) {
+    LOG(message_group::Error, "No application configured, check Preferences -> 3D Print -> Local Application");
+    return false;
+  }
+
 #ifdef Q_OS_MACOS
-  if(!process.startDetached("open", {"-a", slicer, QString::fromStdString(exportedFilename_)})) {
-#else	  
-  if(!process.startDetached(slicer, {QString::fromStdString(exportedFilename_)})) {
+  if (!process.startDetached("open", {"-a", application, QString::fromStdString(exportedFilename_)})) {
+#else
+  QStringList args;
+  const auto info = QFileInfo(QString::fromStdString(exportedFilename_));
+  for (const auto& arg : Settings::Settings::localAppParameterList.items()) {
+    switch (arg.type) {
+    case Settings::LocalAppParameterType::string:
+      args.append(QString::fromStdString(arg.value));
+      break;
+    case Settings::LocalAppParameterType::file:
+      args.append(QString::fromStdString(exportedFilename_));
+      break;
+    case Settings::LocalAppParameterType::dir:
+      args.append(info.absoluteDir().path());
+      break;
+    case Settings::LocalAppParameterType::extension:
+      args.append(info.suffix());
+      break;
+    case Settings::LocalAppParameterType::source:
+      args.append(QString::fromStdString(sourceFilename_));
```

Comment:
Since we have the option to only supply the source file, there isn't technically a need to render before sending this to an external app.
This might be out of scope for this PR though.

Reply:
Yes, I know. It's never safe to assume we have a source file on disk. We could leave the 2 source options out, I was not really sure if it's a good idea to include.

Reply:
I say leave it for now - we can let it simmer in public for a bit